app/javascript/app/components/text-input/text-input-component.jsx

```diff
@@ -1,33 +1,87 @@
-import React from 'react';
+import React, { Component } from 'react';
 import { themr } from 'react-css-themr';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 
 import styles from './text-input-styles';
 
-const InputComponent = ({
-  className,
-  value,
-  theme,
-  onChange,
-  onFocus,
-  failed
-}) => (
-  <input
-    onChange={e => onChange(e.target.value)}
-    onFocus={e => onFocus(e.target.value)}
-    className={cx(className, theme.input, { [theme.inputFailed]: failed })}
-    type="text"
-    value={value}
-  />
-);
+class InputComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: props.value
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
It will be interesting to read this article: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html
Also, componentWillReceiveProps will be deprecated at some point (React 17 as it seems) and in react 16.4 getDerivedStateFromProps is preferred. No big deal right now as we are using 16.2. But maybe something to think about in the future